processing/src/main/java/org/apache/druid/query/filter/vector/VectorValueMatcherFactory.java

```diff
@@ -27,8 +27,14 @@
 
 public interface VectorValueMatcherFactory
 {
+  /**
+   * Specialized value matcher for string equality
+   */
   VectorValueMatcher makeMatcher(@Nullable String value);
 
+  /**
+   * Specialized value matcher for equality
```

Comment:
for what kind of equality? Javadoc should call out when this method would be used. (Otherwise it's hard for implementers to know what kind of `Object` might show up.)